portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,70 +27,107 @@
+    /** One time initialization. */
     @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        List<String> funFacts = new ArrayList<>();
+    public void init() {
         funFacts.add("I enjoy all music genres especially EDM!");
         funFacts.add("I studied abroad in Barcelona :)");
         funFacts.add("My favorite food is rotisserie chicken!");
         funFacts.add("I spend a lot of days on the lake!");
+    }
+ 
+    /** Merges dataFactsAndComments with pastComments to convert as one ArrayList into JSON to put on servlet. */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {        
+        // Add fun facts first to factsAndComments return variable.
+        List<String> factsAndComments = new ArrayList<>();
+        factsAndComments.addAll(funFacts);        
         
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Query query = new Query("siteComments").addSort("timeStamp", SortDirection.DESCENDING);
         PreparedQuery results = datastore.prepare(query);
-
+ 
         List<String> pastComments = new ArrayList<>();
+        int validNumRequest = 0;
+        int numCommentsWanted = 0;
+        boolean numCommentsWantedTrigger = true;
+ 
         for (Entity entity : results.asIterable()) {
-            String comment = (String) entity.getProperty("newComment");
-            pastComments.add(comment);
+            validNumRequest = Integer.parseInt((String)(entity.getProperty("seeNumComments")));
+            if (validNumRequest >= 0){
+                String comment = (String) (entity.getProperty("newComment"));
+
+                //For testing queried result comments
+                // System.out.println("Query result comment: " + comment);
+
+                pastComments.add(comment);
+            } else {
+                System.out.println("Input of number comments wanted not greater than or equal to 0: " + validNumRequest);
+            }
+           
+            //Grab number of comments wanted to be seen by user only in the first entity
+            while (numCommentsWantedTrigger) {
+                numCommentsWanted = validNumRequest;
+                numCommentsWantedTrigger = false;
+            }
+        }
+ 
+        if (numCommentsWanted > pastComments.size()) {
+            numCommentsWanted = pastComments.size();           
         }
+ 
+        pastComments = pastComments.subList(0, numCommentsWanted);
 
-        List<String> factsAndComments =  new ArrayList<>();
-        factsAndComments.addAll(funFacts);
-        factsAndComments.addAll(pastComments);
+        // For testing proper number and right comments are in sublist
+        // for (String comment : pastComments) {
+        //     System.out.println("sublist comment: " + comment);
+        // }
```

Comment:
Again, might want to correct the spacing for these comments (for code, don't have a space after "//").

(We've got a code style bot internally that detects this sort of thing in our code)

Reply:
I'm assuming we can still have the tab for indented code lines

Reply:
Yeah absolutely!  Keep indentations. :)